Foundation/NSArray.swift

```diff
@@ -219,15 +201,15 @@ open class NSArray : NSObject, NSCopying, NSMutableCopying, NSSecureCoding, NSCo
         var descriptions = [String]()
         let cnt = count
         for idx in 0..<cnt {
-            let obj = self[idx] as! NSObject
-            if let string = obj as? NSString {
-                descriptions.append(string._swiftObject)
-            } else if let array = obj as? NSArray {
-                descriptions.append(array.description(withLocale: locale, indent: level + 1))
-            } else if let dict = obj as? NSDictionary {
-                descriptions.append(dict.description(withLocale: locale, indent: level + 1))
+            let obj = self[idx]
+            if let string = obj as? String {
+                descriptions.append(string)
+            } else if let array = obj as? [Any] {
+                descriptions.append(NSArray(array: array).description(withLocale: locale, indent: level + 1))
```

Comment:
This seems inefficient — is there a way to improve on it, or was this change necessary because of type/bridging changes?